cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractContext.java

```diff
@@ -101,13 +101,13 @@ protected AbstractContext(Program program, ProgramOptions programOptions,
                             @Nullable MetricsCollectionService metricsService, Map<String, String> metricsTags,
                             SecureStore secureStore, SecureStoreManager secureStoreManager,
                             @Nullable PluginInstantiator pluginInstantiator) {
-    super(program.getId().toEntityId());
+    super(program.getId());
 
     this.program = program;
     this.programOptions = programOptions;
     this.runId = ProgramRunners.getRunId(programOptions);
     this.discoveryServiceClient = discoveryServiceClient;
-    this.owners = createOwners(program.getId());
+    this.owners = createOwners(program.getId().toId());
```

Comment:
if ``createOwners`` is a private method, change it to accept ``ProgramId`` instead of creating these multiple objects?